### PR TITLE
Use the selected ITC result rather than the first

### PR DESCRIPTION
**Pull request by cquiroz**
This uses the new lucuma-itc API to get the preferred result and also apply limits indicated on

https://app.shortcut.com/lucuma/story/1999/determine-exposure-time-for-acquisition-images#activity-2516

### CODE REVIEW

File: modules/service/src/main/scala/lucuma/odb/service/ItcService.scala [hunk 1]
```diff
@@ -384,17 +384,18 @@ object ItcService {
...
           .map(_.result)
           .recover {
             case ResponseException(errors, _) if errors.exists(_.message.contains("target is too bright")) =>
               // Use default if target is too bright
               Acquisition.DefaultIntegrationTime
           }.map {
-            case r if r.head.exposureTime > Acquisition.MaxExposureTime => r.map(_.copy(exposureTime = Acquisition.MaxExposureTime))
-            case r if r.head.exposureTime < Acquisition.MinExposureTime => r.map(_.copy(exposureTime = Acquisition.MinExposureTime))
+            case r if r.focus.exposureTime > Acquisition.MaxExposureTime => r.map(_.copy(exposureTime = Acquisition.MaxExposureTime))
+            case r if r.focus.exposureTime < Acquisition.MinExposureTime => r.map(_.copy(Acquisition.MaxExposureTime))
```
Comment (toddburnside) [inline modules/service/src/main/scala/lucuma/odb/service/ItcService.scala:396]: Is this correct?

**Review by toddburnside — APPROVED** [review]
LGTM

File: modules/service/src/main/scala/lucuma/odb/service/ItcService.scala (hunk 1 above)
Comment (cquiroz) [reply modules/service/src/main/scala/lucuma/odb/service/ItcService.scala:396]: Good catch

File: modules/sequence/src/main/scala/lucuma/odb/sequence/gmos/longslit/Acquisition.scala [hunk 2]
```diff
@@ -68,7 +79,7 @@ sealed trait Acquisition[D, G, F, U] extends SequenceState[D] {
         _  <- optics.roi           := GmosRoi.CentralStamp
         s1 <- scienceStep(10.arcsec, 0.arcsec, ObserveClass.Acquisition)
 
-        _  <- optics.exposure      := (exposureTime * NonNegInt.unsafeFrom(3)).timeSpan
+        _  <- optics.exposure      := lastExpTime(exposureTime)
```
Comment (swalker2m) [inline modules/sequence/src/main/scala/lucuma/odb/sequence/gmos/longslit/Acquisition.scala:82]: Is it okay if the first step exposure time is over 360?

Comment (cquiroz) [reply modules/sequence/src/main/scala/lucuma/odb/sequence/gmos/longslit/Acquisition.scala:82]: The requirement was stated as
`Could we limit the exposure time of step 3 to <= 360 sec?`

**Review by swalker2m — APPROVED** [review]
Looks great.  :+1:

**Pull request merged**